app/src/main/java/io/appium/uiautomator2/handler/NewSession.java

```diff
@@ -30,4 +37,19 @@ public AppiumResponse safeHandle(IHttpRequest request) {
         return new AppiumResponse(sessionID, WDStatus.SUCCESS, "Created Session");
     }
 
+
+    public  Map<String, Object> getCapabilities(IHttpRequest request) throws JSONException {
+        JSONObject caps = getPayload(request).getJSONObject("desiredCapabilities");
+        Map<String, Object> map = new HashMap<String, Object>();
+
```

Comment:
Instead of using iterator, get and next, can we use below code

```
for (Map.Entry<String, Object> entry : caps.entrySet()) {
   map.put(entry.getKey(), entry.getValue());
}
```